libraries/shared/repository/repository.go

```diff
@@ -16,128 +16,12 @@
-				WHERE (header_id ISNULL OR ` + notCheckedSQL + `)
-				AND headers.block_number >= $1
-				AND headers.eth_node_fingerprint = $2`
-		err = db.Select(&result, query, startingBlockNumber, db.Node.ID)
-	} else {
-		query = `SELECT headers.id, headers.block_number, headers.hash FROM headers
-				LEFT JOIN checked_headers on headers.id = header_id
-				WHERE (header_id ISNULL OR ` + notCheckedSQL + `)
-				AND headers.block_number >= $1
-				AND headers.block_number <= $2
-				AND headers.eth_node_fingerprint = $3`
-		err = db.Select(&result, query, startingBlockNumber, endingBlockNumber, db.Node.ID)
-	}
-
-	return result, err
-}
-
-func GetCheckedColumnNames(db *postgres.DB) ([]string, error) {
-	// Query returns `[]driver.Value`, nullable polymorphic interface
-	var queryResult []driver.Value
-	columnNamesQuery :=
-		`SELECT column_name FROM information_schema.columns
-		WHERE table_schema = 'public'
-			AND table_name = 'checked_headers'
-			AND column_name <> 'id'
-			AND column_name <> 'header_id';`
-
-	err := db.Select(&queryResult, columnNamesQuery)
-	if err != nil {
-		return []string{}, err
-	}
-
-	// Transform column names from `driver.Value` to strings
-	var columnNames []string
-	for _, result := range queryResult {
-		if columnName, ok := result.(string); ok {
-			columnNames = append(columnNames, columnName)
-		} else {
-			return []string{}, fmt.Errorf("incorrect value for checked_headers column name")
-		}
-	}
-
-	return columnNames, nil
-}
-
-// Builds a SQL string that checks if any column should be checked/rechecked.
-// Defaults to FALSE when no columns are provided.
-// Ex: ["columnA", "columnB"] => "NOT (columnA!=0 AND columnB!=0)"
-//     [] => "FALSE"
-func CreateHeaderCheckedPredicateSQL(boolColumns []string, recheckHeaders constants.TransformerExecution) string {
-	if len(boolColumns) == 0 {
-		return "FALSE"
-	}
-
-	if recheckHeaders {
-		return createHeaderCheckedPredicateSQLForRecheckedHeaders(boolColumns)
-	} else {
-		return createHeaderCheckedPredicateSQLForMissingHeaders(boolColumns)
-	}
-}
-
-func createHeaderCheckedPredicateSQLForMissingHeaders(boolColumns []string) string {
-	var result bytes.Buffer
-	result.WriteString(" (")
-
-	// Loop excluding last column name
-	for _, column := range boolColumns[:len(boolColumns)-1] {
-		result.WriteString(fmt.Sprintf("%v=0 OR ", column))
-	}
-
-	result.WriteString(fmt.Sprintf("%v=0)", boolColumns[len(boolColumns)-1]))
-
-	return result.String()
-}
-
-func createHeaderCheckedPredicateSQLForRecheckedHeaders(boolColumns []string) string {
-	var result bytes.Buffer
-	result.WriteString(" (")
-
-	// Loop excluding last column name
-	for _, column := range boolColumns[:len(boolColumns)-1] {
-		result.WriteString(fmt.Sprintf("%v<%s OR ", column, constants.RecheckHeaderCap))
-	}
-
-	// No trailing "OR" for the last column name
-	result.WriteString(fmt.Sprintf("%v<%s)", boolColumns[len(boolColumns)-1], constants.RecheckHeaderCap))
-
-	return result.String()
```

Comment:
Nice to finally nuke this abomination :sweat_smile:  How do we ensure the same functionality though?

Reply:
Define "same". I think we're ensuring that we've checked for all _currently configured_ transformers - but probably do need to think more about how to support knowing you haven't checked for logs from a _newly added_ transformer after the system has been running for awhile...